internal/handlers/common/params_test.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMultiplyLongSafely(t *testing.T) {
+	t.Parallel()
+
+	for name, tc := range map[string]struct {
+		values []int64 // [v1,v2,v1+v2] or [v1,v2] if you don't want to check multiplication result
+
+		// overflow checks if at least one value returned errLongExceeded error.
+		overflow bool // defaults to false
+
+		// valuesRange allows to execute test for the numbers next to v1.
+		// if v1=42 and valuesRange=2, following numbers will be tested: [42,43,44]
+		// It shouldn't be used with multiplication result, but rather to test overflows on large ranges.
+		valuesRange int
```

Comment:
This one is useful if we want to check for overflow on large values. We haven't realized that we have a bug here so far, because we've used one or two single big values which were able to be presented as float64 precisely. While all other numbers around those values were loosing their precision.

Reply:
Fixed, I picked the value from the range that might be rounded because of double precision